### PR TITLE
Apply minor polish to ClassInclusionReport

### PR DESCRIPTION
**Pull request by geoand**
@Sanne introduced this very useful utility that can come in
handy for integration testing.
This change applies some minor polish, the most important
being to guard against a potential NPE

### CODE REVIEW

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/nativeimage/ClassInclusionReport.java [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
     private final Set<String> includedClasses;
 
-    private ClassInclusionReport(Set<String> strings, Path usedClassesReport) {
+    private ClassInclusionReport(Set<String> strings) {
```
Comment (Sanne) [inline test-framework/junit5/src/main/java/io/quarkus/test/junit/nativeimage/ClassInclusionReport.java:22]: ah I see this wasn't complete, but I actually meant to store the `usedClassedReport` so to not access filesystem multiple times. Wouldn't it be better?

Comment (geoand) [reply test-framework/junit5/src/main/java/io/quarkus/test/junit/nativeimage/ClassInclusionReport.java:22]: Sure yeah, I'll do it :)

Comment (geoand) [reply test-framework/junit5/src/main/java/io/quarkus/test/junit/nativeimage/ClassInclusionReport.java:22]: Done

**Review by Sanne — APPROVED** [review]
Thanks!

**Pull request merged**